src/lib/mina.ts

```diff
@@ -705,13 +717,20 @@ function Network(
       Fetch.setGraphqlEndpoint(minaGraphqlEndpoint);
     }
 
-    if (Array.isArray(input.archive) && input.archive.length !== 0) {
-      archiveEndpoint = input.archive[0];
-      Fetch.setArchiveGraphqlEndpoint(archiveEndpoint);
-      Fetch.setArchiveGraphqlFallbackEndpoints(input.archive.slice(1));
-    } else if (typeof input.archive === 'string') {
-      archiveEndpoint = input.archive;
-      Fetch.setArchiveGraphqlEndpoint(archiveEndpoint);
+    if (input.archive) {
+      if (Array.isArray(input.archive) && input.archive.length !== 0) {
+        archiveEndpoint = input.archive[0];
+        Fetch.setArchiveGraphqlEndpoint(archiveEndpoint);
+        Fetch.setArchiveGraphqlFallbackEndpoints(input.archive.slice(1));
+      } else if (typeof input.archive === 'string') {
+        archiveEndpoint = input.archive;
+        Fetch.setArchiveGraphqlEndpoint(archiveEndpoint);
+      }
+    }
+
+    if (input.accountsManager && typeof input.accountsManager === 'string') {
```

Comment:
please use `if (input.accountsManager !== undefined)`. We don't use runtime typeguards since we rely on TS (`accountsManager` can only be `string | undefined`)

Reply:
hey @shimkiv here I also meant remove the `typeof input.accountsManager === 'string'` check. we don't do checks like that since the type enforces it already.